src/hooks/useEventCallback.ts

```diff
@@ -1,14 +1,15 @@
-import { useRef, useEffect, useCallback } from "react";
+import { useRef } from "react";
 
+function noop() {}
 // Saves incoming handler to the ref in order to avoid "useCallback hell"
 function useEventCallback<T>(handler?: (value: T) => void): (value: T) => void {
   const callbackRef = useRef(handler);
 
-  useEffect(() => {
```

Comment:
There is no need in pause, when a new function was passed, it doesn't have any async side effects, therefore removing useEffect makes the performance better

Reply:
Smart!

Reply:
@jeetiss check this out. You might find this approach interesting.